core/encoding_jbig2.go

```diff
@@ -0,0 +1,437 @@
+func (enc *JBIG2Encoder) EncodeImage(img image.Image) ([]byte, error) {
+	return enc.encodeImage(img)
+}
+
+// Encode encodes previously prepare jbig2 document and stores it as the byte slice.
+func (enc *JBIG2Encoder) Encode() (data []byte, err error) {
+	const processName = "JBIG2Document.Encode"
+	if enc.d == nil {
+		return nil, errors.Errorf(processName, "document input data not defined")
+	}
+	enc.d.FullHeaders = enc.DefaultPageSettings.FileMode
+	// encode the document
+	data, err = enc.d.Encode()
+	if err != nil {
+		return nil, errors.Wrap(err, processName, "")
+	}
+	return data, nil
+}
+
+// GetFilterName returns the name of the encoding filter.
+func (enc *JBIG2Encoder) GetFilterName() string {
+	return StreamEncodingFilterNameJBIG2
+}
+
+// MakeDecodeParams makes a new instance of an encoding dictionary based on the current encoder settings.
+func (enc *JBIG2Encoder) MakeDecodeParams() PdfObject {
+	return MakeDict()
+}
+
+// MakeStreamDict makes a new instance of an encoding dictionary for a stream object.
+func (enc *JBIG2Encoder) MakeStreamDict() *PdfObjectDictionary {
+	dict := MakeDict()
+	dict.Set("Filter", MakeName(enc.GetFilterName()))
+	return dict
+}
+
+// UpdateParams updates the parameter values of the encoder.
+func (enc *JBIG2Encoder) UpdateParams(params *PdfObjectDictionary) {
+}
+
+func (enc *JBIG2Encoder) encodeImage(i image.Image) ([]byte, error) {
+	const processName = "encodeImage"
+	// convert the input into jbig2 image
+	jbig2Image, err := GoImageToJBIG2(i, JB2ImageAutoThreshold)
+	if err != nil {
+		return nil, errors.Wrap(err, processName, "convert input image to jbig2 img")
+	}
+	if err = enc.AddPageImage(jbig2Image, &enc.DefaultPageSettings); err != nil {
+		return nil, errors.Wrap(err, processName, "")
+	}
+	return enc.Encode()
+}
+
+func newJBIG2DecoderFromStream(streamObj *PdfObjectStream, decodeParams *PdfObjectDictionary) (*JBIG2Encoder, error) {
+	const processName = "newJBIG2DecoderFromStream"
+	encoder := &JBIG2Encoder{}
+	encDict := streamObj.PdfObjectDictionary
+	if encDict == nil {
+		// No encoding dictionary.
+		return encoder, nil
+	}
+
+	// If decodeParams not provided, see if we can get from the stream.
+	if decodeParams == nil {
+		obj := encDict.Get("DecodeParms")
+		if obj != nil {
+			switch t := obj.(type) {
+			case *PdfObjectDictionary:
+				decodeParams = t
+			case *PdfObjectArray:
+				if t.Len() == 1 {
+					if dp, ok := GetDict(t.Get(0)); ok {
+						decodeParams = dp
+					}
+				}
+			default:
+				common.Log.Error("DecodeParams not a dictionary %#v", obj)
+				return nil, errors.Errorf(processName, "invalid DecodeParms type: %T", t)
+			}
+		}
+	}
+
+	if decodeParams != nil {
+		if globals := decodeParams.Get("JBIG2Globals"); globals != nil {
+			var err error
+
+			globalsStream, ok := globals.(*PdfObjectStream)
+			if !ok {
+				err = errors.Error(processName, "jbig2.Globals stream should be an Object Stream")
+				//noinspection GoNilness
```

Comment:
Can you remove those noinspection comments?
At the moment we just try to minimize golint output but not commenting on each single deviation.